File: Swift/L4/Lista_04_EX03/Lista_04_EX03/Produto.swift
//  Produto.swift
//  Lista_04_EX03

import Foundation

public class Produto {
    
    private var nome: String
    private var preco: Double
    private var imposto: ImpostoEnum
    

    init(nome: String, preco: Double, imposto: ImpostoEnum) {
        self.nome    = nome
        self.preco   = preco
        self.imposto = imposto
    }
    
    public func mostraProduto() {
    print("*********************************")
    print("Nome do Produto....: \(self.nome)")
    print("Tipo do Imposto....: \(self.imposto)")
    print("Preço sem Imposto..: R$ \(self.preco)")
    print("Alíquota do Imposto: \( (self.imposto.rawValue - 1)*100) %")
    print("Preço com Imposto..: R$ \( (( self.imposto.rawValue * self.preco * 100).rounded())/100)" )
    print("*********************************\n")
    }//fim func mostra produto
    
}//fim class produto

File: Lista_04_EX03/src/Produto.java

/* Exercício 1.3. Um produto é composto por nome e preço.
 * Sabe-se que todo produto tem um imposto apenas.
 * IPI e ICMS são impostos possíveis
 * (um produto pode ter apenas um deles para facilitar).
 * IPI e ICMS possuem aliquotas de 5% e 10% (valores ﬁctícios) respectivamente.
 * Todo produto deve ter a funcionalidade de mostrar
 * o nome e o preço já descontado o imposto.
 * Use o máximo de orientação a objetos possível para codiﬁcar as classes.
 */

public class Produto {
	
	private String nome;
	private double preco;
	private ImpostoEnum imposto;
	
	
	public Produto(String nome, double preco, ImpostoEnum imposto) {
		super();
		this.nome = nome;
		this.preco = preco;
		this.imposto = imposto;
	}//fim construtor
	
	
	public void mostraProduto() {
	System.out.println("*********************************");
	System.out.println("Nome do Produto....: " + this.nome);
	System.out.println("Tipo do Imposto....: " + this.imposto);
	System.out.printf( "Preço Normal.......: R$ %.2f %n", this.preco);
	System.out.printf( "Alíquota do Imposto: %.2f", ( ( this.imposto.getImposto()-1.0 ) *100.0));
	System.out.println(" %");
	System.out.printf( "Preço com Imposto..: R$ %.2f %n", (this.preco * this.imposto.getImposto()));
	System.out.println("*********************************");
	}//fim metodo mostra produto
	
}
